src/text/line.rs

```diff
@@ -100,14 +100,22 @@ impl<'a> Line<'a> {
 }
 
 impl<'a> From<String> for Line<'a> {
+    #[inline]
     fn from(s: String) -> Self {
-        Self::from(vec![Span::from(s)])
+        Self::from(match s.as_str() {
+            "" => vec![],
+            _ => vec![Span::from(s)],
+        })
```

Comment:
I looked over all the code that uses `Line`s, and couldn't find any code that would appear to be affected. The existing tests didn't either. It's possible something slipped by us both, of course.

Reply:
As for the reason I touched this code at all, it's because empty lines are likely to be produced more often with this PR (see explanation in earlier reply). So I wanted to ensure they were properly optimized.

Reply:
Would it make sense to you if we saved optimizing for when there's a real need backed by benchmarks that suggest those parts are hot and heavy. I'm a big proponent of preferring clarity and only moving away from that when necessary. I wouldn't be surprised if there's no performance improvement at all from this change (or one that's very small).

Reply:
Making this method simpler again would actually require the unit tests for the `fmt::Write` impl become more complex. The derived `PartialEq` impl on `Text` treats two `Text`s as unequal even if they differ only in empty `Span`s, so producing such empty spans would require the tests to model them as well.

Reply:
I wonder if empty spans have a real reason for existing. Perhaps they could or should in future be relevant to choosing where to wrap?

Either way, it's probably worthwhile making the choice of what to do with empty spans explicit. That means regardless of whether we drop or keep them:
- we should do so for all cases, not just the case where there's a single empty span on the line
- there should be tests that define how this happens
- there should be documentation on the methods where this happens that states that this is the behavior

This might sound a bit like over engineering, but because Text really is a core type, many apps and libraries outside of this library would use it and have expectations on the behavior. Dropping empty spans probably doesn't break any use case, but I'm not 100% sure without going and doing a search of every crate that depends on ratatui/tui. Changing that is probably fine, but we'd want to do so intentionally (i.e be consistent / documented / tested).